core/src/main/java/de/danoeh/antennapod/core/dialog/DisplayConfirmationDialog.java

```diff
@@ -0,0 +1,33 @@
+package de.danoeh.antennapod.core.dialog;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.DialogInterface;
+import androidx.annotation.StringRes;
+import java.util.concurrent.Callable;
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+
+public class DisplayConfirmationDialog {
```

Comment:
This feels a bit weird. We already have the `ConfirmationDialog` class. Anyway, when working on the thing I suggested above (`FeedMenuHandler.onMenuItemClicked`), this is no longer needed.

Reply:
I would switch back to the old direct use of `ConfirmationDialog` and pass a Runnable that then gets called when something changes. Then you don't need the new `DisplayConfirmationDialog` class, and not the new `StatusListener` class.